internal/azure/exchanger.go

```diff
@@ -0,0 +1,81 @@
+package azure
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type tokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	Resource     string `json:"resource"`
+	TokenType    string `json:"token_type"`
+}
+
+type Exchanger struct {
+	scheme   string
+	acrFQDN  string
+	tenantID string
+}
+
+func NewExchanger(acrEndpoint string) *Exchanger {
+	return &Exchanger{
+		scheme:  "https",
+		acrFQDN: acrEndpoint,
+	}
+}
+
+func (e *Exchanger) ExchangeACRAccessToken(armToken string) (string, error) {
+	exchangeUrl := fmt.Sprintf("%s://%s/oauth2/exchange", e.scheme, e.acrFQDN)
+	parsedURL, err := url.Parse(exchangeUrl)
+	if err != nil {
+		return "", err
+	}
+
+	parameters := url.Values{}
+	parameters.Add("grant_type", "access_token")
+	parameters.Add("service", parsedURL.Hostname())
+	parameters.Add("access_token", armToken)
+
+	req, err := http.NewRequest("POST", exchangeUrl, strings.NewReader(parameters.Encode()))
+	if err != nil {
+		return "", fmt.Errorf("error constructing exchange request: %w", err)
+	}
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Content-Length", strconv.Itoa(len(parameters.Encode())))
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
```

Comment:
There's a helper method that can be used for form POSTing in `http`. Here's the short variant of all of this :smile: 
```suggestion
	resp, err := http.PostForm(exchangeUrl, parameters)
```